src/ast/ast_build_filter_tree.c

```diff
@@ -296,7 +296,10 @@ void AST_ConvertFilters(FT_FilterNode **root, const cypher_astnode_t *entity) {
 		node = _convertBinaryOperator(entity);
 	} else if(type == CYPHER_AST_UNARY_OPERATOR) {
 		node = _convertUnaryOperator(entity);
-	} else if(type == CYPHER_AST_APPLY_OPERATOR) {
+	} else if(type == CYPHER_AST_APPLY_OPERATOR ||
+			  type == CYPHER_AST_ANY            ||
+			  type == CYPHER_AST_ALL            ||
+			  type == CYPHER_AST_LIST_COMPREHENSION) {
```

Comment:
`CYPHER_AST_LIST_COMPREHENSION` doesn't returns a boolean
Consider: `WITH 1 AS x WHERE [a IN [1,2,3] WHERE a = 5] RETURN x`
The filter expression returns an empty array, from the filter perspective the criteria isn't false as such `x` is returned,

Running this query in NEO yields an empty result-set.